bld/CLMBuildNamelist.pm

```diff
@@ -768,6 +777,31 @@ sub setup_cmdl_fates_mode {
   }
 }
 
+#-------------------------------------------------------------------------------
+sub setup_cmdl_configuration_and_structure {
+   # Error-check and set the 'configuration' and 'structure' namelist flags
+
+   my ($opts, $nl_flags, $definition, $defaults, $nl) = @_;
+
+   my $val;
+
+   my $var = "configuration";
+   $val = $opts->{$var};
+   if ($val eq "clm" || $val eq "nwp") {
```

Comment:
@billsacks, when you invoke warnings perl doesn't like the use of variable that isn't defined (or explicitly set to undef). Sp it's better to do it this way...

`if ( defined($val) && ($val eq "clm" || $val eq "nwp") ) {`

There are other examples of that throughout.

Reply:
Same thing below with structure

Reply:
Done